File: cashierApp/src/main/java/alimentation/cashierApp/services/ProductTypeService.java
package alimentation.cashierApp.services;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;

import alimentation.cashierApp.dao.ProductTypeRepository;
import alimentation.cashierApp.exception.CashierAppException;
import alimentation.cashierApp.models.ProductType;

@Service
public class ProductTypeService {

    @Autowired
    private ProductTypeRepository ProductTypeRepository;


    // Add methods for ProductType-related operations here
    public Iterable<ProductType> getAllProductTypes(){
        return ProductTypeRepository.findAll();
    }

    public Iterable<ProductType> getAllProductTypesByProductType(String productType){
        return ProductTypeRepository.findAllByProductType(productType);
    }

    public ProductType getProductTypeById(int id){
        Optional<ProductType> response = ProductTypeRepository.findById(id);
        if (response.isPresent()) {
            return response.get();
        }
        return null;
    }


    public ProductType getProductTypeByName(String name){
        return ProductTypeRepository.findByName(name);
    }

    public ProductType addProductType(ProductType ProductType){
        return ProductTypeRepository.save(ProductType);
    }

    public ProductType updateProductType(ProductType ProductType){
        return ProductTypeRepository.save(ProductType);
    }

    public ProductType deleteProductType(int id){
        Optional<ProductType> target = ProductTypeRepository.findById(id);
        ProductType result = target.get();
        if (result == null){
            throw new CashierAppException(HttpStatus.BAD_REQUEST, "ProductType missing");
        }
        ProductTypeRepository.deleteById(id);
        return result;
    }

    public void deleteAllProductTypes() {
        ProductTypeRepository.deleteAll();
    }

}

File: cashierAppFront/src/services/ProductTypeService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/productTypes';

class ProductTypeService {
  getAllProductTypes() {
    return axios.get(`${API_URL}/all`);
  }

  getProductTypeById(id) {
    return axios.get(`${API_URL}`, { params: { id } });
  }

  getAllProductTypesByProductType(productType) {
    return axios.get(`${API_URL}/productType`, { params: { productType } });
  }

  createProductType(productTypeDto) {
    return axios.post(API_URL, productTypeDto);
  }

  updateProductType(productTypeDto) {
    return axios.put(API_URL, productTypeDto);
  }

  deleteProductType(id) {
    return axios.delete(`${API_URL}`, { params: { id } });
  }

  deleteAllProductTypes() {
    return axios.delete(`${API_URL}/all`);
  }
}

export default new ProductTypeService();